samples/teacherNotebookA.dib

```diff
@@ -1,22 +1,43 @@
 #!markdown
 
-# Stage 1 (students wont see this line)
+# Stage 1
 
 #!markdown
 
-## [Set up] (students wont see this line)
+## [SetUp]
 
 #!csharp
 
 double GetConstant() => 1.0;
 
+#!csharp
+
+Lesson.NextEvaluation.AddRule(
+    (context) => {
+        context.Code.Contains(4);
+
+        context.ReturnValue.Should().Be(123);
+
+        context.ResultEvents.ContainSingle<CommandSucceeded>();
+    }
+);
+
+#!csharp
+
+Lesson.Evaluation.AddCustomMessaging(
```

Comment:
The messaging should just be part of the rule definition.

Each should be able to specify associated data for when it fails.

Reply:
Makes sense. How would they attach custom messaging if they wanted to use our API out of the box?

Reply:
I've added some examples